streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -63,87 +78,123 @@ public InternalTopicManager(final StreamsKafkaClient streamsKafkaClient,
+                        log.warn("Could not get number of partitions for topic {}. Topic might not exists: {}",
+                            topicFuture.getKey(),
+                            topicDoesNotExist.getMessage());
+                    } else {
+                        throw topicDoesNotExist;
+                    }
+                }
             }
-            // backoff
-            time.sleep(100L);
+
+            return existingTopicPartitions;
+        } catch (final Exception fatalException) {
+            final String error = "Could not get number of partitions from brokers. " +
+                "This can happen if the Kafka cluster is temporary not available. " +
+                "You can increase admin client config `retries` to be resilient against this error.";
+            log.error(error, fatalException);
+            throw new StreamsException(error, fatalException);
         }
-        throw new StreamsException("Could not get number of partitions.");
     }
 
     public void close() {
-        streamsKafkaClient.close();
+        adminClient.close();
     }
 
     /**
      * Check the existing topics to have correct number of partitions; and return the non existing topics to be created
      */
-    private Map<InternalTopicConfig, Integer> validateTopicPartitions(final Map<InternalTopicConfig, Integer> topicsPartitionsMap,
-                                                                      final Map<String, Integer> existingTopicNamesPartitions) {
-        final Map<InternalTopicConfig, Integer> topicsToBeCreated = new HashMap<>();
-        for (Map.Entry<InternalTopicConfig, Integer> entry : topicsPartitionsMap.entrySet()) {
-            InternalTopicConfig topic = entry.getKey();
-            Integer partition = entry.getValue();
+    private Set<InternalTopicConfig> validateTopicPartitions(final Collection<InternalTopicConfig> topicsPartitionsMap,
+                                                             final Map<String, Integer> existingTopicNamesPartitions) {
+        final Set<InternalTopicConfig> topicsToBeCreated = new HashSet<>();
+        for (final InternalTopicConfig topic : topicsPartitionsMap) {
+            final Integer numberOfPartitions = topic.numberOfPartitions();
             if (existingTopicNamesPartitions.containsKey(topic.name())) {
-                if (!existingTopicNamesPartitions.get(topic.name()).equals(partition)) {
-                    throw new StreamsException("Existing internal topic " + topic.name() + " has invalid partitions." +
-                            " Expected: " + partition + " Actual: " + existingTopicNamesPartitions.get(topic.name()) +
-                            ". Use 'kafka.tools.StreamsResetter' tool to clean up invalid topics before processing.");
+                if (!existingTopicNamesPartitions.get(topic.name()).equals(numberOfPartitions)) {
+                    log.error("Existing internal topic {} has invalid partitions: expected: {}; actual: {}. " +
```

Comment:
maybe use `String.format` first and use for both the `log.error` and the exception message